gradle/src/main/groovy/com/flurry/android/symbols/NdkSymbolUpload.groovy

```diff
@@ -50,10 +50,14 @@ class NdkSymbolUpload {
         }
 
         Closure searchForSharedObjectFiles = { ExternalNativeBuildTask task ->
-            File objFolder = task.objFolder
-            File soFolder = task.soFolder
-            findSharedObjectFiles(objFolder, addSharedObjectFiles)
-            findSharedObjectFiles(soFolder, addSharedObjectFiles)
+            if (task.objFolder instanceof DirectoryProperty) {
+               findSharedObjectFiles(task.objFolder.asFileTree.files, addSharedObjectFiles)
+               findSharedObjectFiles(task.soFolder.asFileTree.files, addSharedObjectFiles)
+           }
+           else {
+               findSharedObjectFiles(new HashSet<File>(Set.of(task.objFolder)), addSharedObjectFiles)
+               findSharedObjectFiles(new HashSet<File>(Set.of(task.soFolder)), addSharedObjectFiles)
```

Comment:
Why are we wrapping `Set.of` in `HashSet` here?

Reply:
agree with @clayreimann that this looks excessive given the signature of `findSharedObjectFiles()`